test/test_macros.hpp

```diff
@@ -8,5 +8,22 @@
 #pragma once
 
 #include <cstdio>
+#include <cfloat>
+#include <math.h>
+
+inline float radians(float deg) {
+    return M_PI / 180.0f * deg;
+}
+
+inline float degrees(float rad) {
+    return rad / (M_PI / 180.0f);
+}
+
+inline bool is_equal(float a, float b, float e) {
+    return fabsf(a - b) < e;
+}
+inline bool is_equal(float a, float b) {
+    return is_equal(a, b, FLT_EPSILON);
+}
 
```

Comment:
Don't use float & in it though. Use just float. I have not checked if you
updated that yet.

On 16 Aug 2016 18:10, "James Goppert" <notifications@github.com> wrote:

> In test/test_macros.hpp
> <https://github.com/PX4/Matrix/pull/26#discussion_r74966654>:
>
> > +#include <math.h>
> > +
> > +inline float radians(float deg) {
> > +    return M_PI / 180.0f * deg;
> > +}
> > +
> > +inline float degrees(float rad) {
> > +    return rad / (M_PI / 180.0f);
> > +}
> > +
> > +inline bool is_equal(float a, float b, float e) {
> > +    return fabsf(a - b) < e;
> > +}
> > +inline bool is_equal(float a, float b) {
> > +    return is_equal(a, b, FLT_EPSILON);
> > +}
> >
>
> I just add a isEqual function to master, it should cover this use case.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/PX4/Matrix/pull/26/files/3a1e1e3d2bdfaebd9efd87bdeb9d135dfe6d4453#r74966654>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AERqiQH2tdRAfz4Z9Owq9Y5coM4ifLLqks5qgeEKgaJpZM4JZZwu>
> .
>